### PR TITLE
Fixes #4754: report other mouse buttons

### LINKED ISSUE #4754: Report other mouse buttons
I've tried map mouse button X1 & X2 in vim, but nothing happened when I clicked. And double click also heppened nothing.

### PR DESCRIPTION
**Pull request by VishnuSanal**
Supersedes #6957

### CODE REVIEW

File: alacritty/src/input/mod.rs [hunk 1]
```diff
@@ -946,7 +956,7 @@ impl<T: EventListener, A: ActionContext<T>> Processor<T, A> {
             MouseButton::Left => self.ctx.mouse_mut().left_button_state = state,
             MouseButton::Middle => self.ctx.mouse_mut().middle_button_state = state,
             MouseButton::Right => self.ctx.mouse_mut().right_button_state = state,
-            _ => (),
+            _ => self.ctx.mouse_mut().other_button_state = state,
```
[the remark this answers is not in the document]
Comment (VishnuSanal) [reply alacritty/src/input/mod.rs:961]: PTAL :)

[the remark this answers is not in the document]
Comment (chrisduerr) [reply alacritty/src/input/mod.rs:961]: The issue is still exactly the same?

**Review by chrisduerr — CHANGES_REQUESTED** [review]

File: alacritty/src/input/mod.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (VishnuSanal) [reply alacritty/src/input/mod.rs:961]: hi, it was specified [here](https://github.com/alacritty/alacritty/pull/6957#issuecomment-1595028204) to only map `Back` and `Forward` to `8` and `9` & to report other buttons according to `xterm`.

this was why only `MouseButton::Back` & `MouseButton::Forward` was handled in the `match` statement. and the default case was to handle the other buttons as is.

hence, we check for the buttons pressed on `#on_mouse_release`, add the respective values & report them as is.

please correct me if I am missing something here, thanks! :)

[the remark this answers is not in the document]
Comment (chrisduerr) [reply alacritty/src/input/mod.rs:961]: > and the default case was to handle the other buttons as is.

That is **not** what your code does.

File: alacritty/src/input/mod.rs [hunk 2]
```diff
@@ -595,8 +595,13 @@ impl<T: EventListener, A: ActionContext<T>> Processor<T, A> {
                 MouseButton::Left => 0,
                 MouseButton::Middle => 1,
                 MouseButton::Right => 2,
-                // Can't properly report more than three buttons..
-                MouseButton::Back | MouseButton::Forward | MouseButton::Other(_) => return,
+                MouseButton::Back => 8,
+                MouseButton::Forward => 9,
+                MouseButton::Other(x) => match x {
+                    6 | 7 => (x + 64) as u8,
+                    10 | 11 => (x + 128) as u8,
```
Comment (chrisduerr) [inline alacritty/src/input/mod.rs:602]: Why did you add this? What is it based on? X11 button IDs? How about cross-platform compatibility?

**Review by chrisduerr — CHANGES_REQUESTED** [review]

File: alacritty/src/input/mod.rs (hunk 2 above)
Comment (VishnuSanal) [reply alacritty/src/input/mod.rs:602]: hi

> Why did you add this? What is it based on? X11 button IDs?

it is based on XTerm control sequences (as mentioned here: https://github.com/alacritty/alacritty/pull/6957#issuecomment-1565190532).

> How about cross-platform compatibility?

I think cross-platform compatibility is already being handled by `winit` (ref: https://github.com/rust-windowing/winit/pull/2770)

Comment (kchibisov) [reply alacritty/src/input/mod.rs:602]: >I think cross-platform compatibility is already being handled by winit (ref: https://github.com/rust-windowing/winit/pull/2770)

This is irrelevant here, because winit guarantee only for `Named` values, but you're mapping `Other` and `Other` is _undefined_. And that's what was the question about here, why you translate `6 | 7` to `x + 64` where on wayland or macOS the buttons are just completely different, thus you kind of implies X11 because xterm is X11.

Comment (VishnuSanal) [reply alacritty/src/input/mod.rs:602]: > > I think cross-platform compatibility is already being handled by winit (ref: [rust-windowing/winit#2770](https://github.com/rust-windowing/winit/pull/2770))
> 
> This is irrelevant here, because winit guarantee only for `Named` values, but you're mapping `Other` and `Other` is _undefined_. And that's what was the question about here, why you translate `6 | 7` to `x + 64` where on wayland or macOS the buttons are just completely different, thus you kind of implies X11 because xterm is X11.

alright, I will look into this & update the PR soon.

**Comment by chrisduerr** [comment]
> I will look into this & update the PR soon.

RIP.
